.circleci/wait-until-previous-builds-complete.sh

```diff
@@ -0,0 +1,167 @@
+#! /bin/bash
+
+set -e
+set -o pipefail
+set -o functrace
+
+#********************
+# VARIABLES
+# *******************
+api_root="https://circleci.com/api/v2/"
+# polling branch name. Other branches (PRs) are excluded from consideration.
+branch="master"
+# Polling workflow name. Other workflow names (nightly-tests and deploy-staging) are excluded from consideration.
+workflow_name="build-test-deploy"
+project_slug="gh/all-of-us"
+
+#********************
+# FUNCTIONS
+# *******************
+
+# CIRCLECI_TOKEN is a env variable whose value is personal token.
+# See https://circleci.com/docs/2.0/managing-api-tokens/#creating-a-personal-api-token
+check_circleci_token() {
+  if [[ ! $CIRCLECI_TOKEN ]]; then
+    printf '%s\n' "Required env variable \"CIRCLECI_TOKEN\" not found.\n Create a personal token then create \"CIRCLECI_TOKEN\" env variable?\n" >&2
+    exit 1
+  fi
+}
+
+post() {
+  local url="${api_root}${1}"
+  printf "curl POST ${url}\n" >/dev/tty
+  curl -X POST \
+    -H "Content-Type: application/json" \
+    -H "Circle-Token: ${CIRCLECI_TOKEN}" \
+    -d "$2" \
+    "${url}"
+}
+
+get() {
+  local url="${api_root}${1}"
+  printf "curl GET ${url}\n" >/dev/tty
+  curl -X GET -sS \
+    -H "Content-Type: application/json" \
+    -H "Circle-Token: ${CIRCLECI_TOKEN}" \
+    "${url}"
+}
+
+# Get list of recently built pipelines. Save results to json file.
+pipeline_json="/tmp/master_branch_pipelines.json"
+fetch_pipeline_ids() {
+  local get_path="pipeline?org-slug=${project_slug}"
+  local get_result=$(get "${get_path}")
+  echo ${get_result} | jq "[.items[] | select(.vcs.branch==\"master\")][] | {created_at: .created_at, id: .id, number: .number}" > "${pipeline_json}"
+  printf "Found following pipelines on ${branch} branch:\n"
+  cat ${pipeline_json}
+  printf "\n"
+}
+
+fetch_pipeline_number() {
+  # Remove double or single quotes.
+  local id=$(echo "${1}" | xargs echo)
+  local get_path="pipeline/${id}"
+  local get_result=$(get "${get_path}")
+  __=$(echo "${get_result}" | jq -r .number)
+}
+
+# https://circleci.com/docs/2.0/workflows/#states
+# v2 workflow api tells status, but not the pipeline api.
+fetch_workflow_status() {
+  # Remove double or single quotes.
+  local id=$(echo "${1}" | xargs echo)
+  local get_path="pipeline/${id}/workflow"
+  local get_result=$(get "${get_path}")
+  local workflow_summary=$(echo "${get_result}" | jq ".items[] | {name: .name, id: .id, status: .status, pipeline_number: .pipeline_number}")
+  printf "${workflow_summary}\n"
+  # workflow branch name is bound by $workflow_name variable.
+  # Rerunning a failed workflow produces a nested datetime sorted array. Get the status of latest run (first array element).
+  __=$(echo "${get_result}" | jq -r "first(.items[]) | select(.name==\"${workflow_name}\") | .status | @sh")
```

Comment:
Like this

<img width="667" alt="Screen Shot 2021-07-16 at 12 47 19 PM" src="https://user-images.githubusercontent.com/35533885/125981822-35032559-fab4-47be-8c26-de0c975e886e.png">